src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -50,31 +45,31 @@ public void Dispose()
         [System.Security.SecuritySafeCritical]  // auto-generated
         private unsafe void Dispose(bool disposing)
         {
-            if (_store != null)
+            if (_stream != null)
             {
 
                 if (disposing)
                 {
                     // Close the stream in a thread-safe way.  This fix means 
                     // that we may call Close n times, but that's safe.
-                    BinaryReader copyOfStore = _store;
-                    _store = null;
-                    if (copyOfStore != null)
-                        copyOfStore.Dispose();
+                    Stream stream = _stream;
```

Comment:
I thought that because the reader is stateful (seeks the stream) using it on multiple threads will not work. But I will do roughly what you proposed above and then do some threadsafety/consistency testing/hardening later.